### PR TITLE
QuizViewController and QuizViewModel with mock data

### PR DESCRIPTION
**Pull request by LorenaL9**
https://user-images.githubusercontent.com/94188461/181747731-f7339401-4fc5-45b9-923b-220f290b7a9f.mp4

### CODE REVIEW

File: QuizApp/Quiz/View/QuizViewController.swift [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+            .store(in: &disposables)
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        quizCategories.collectionViewLayout.invalidateLayout()
+    }
+
+}
+
+extension QuizViewController: ConstructViewsProtocol {
```
Comment (marinzg) [inline QuizApp/Quiz/View/QuizViewController.swift:74]: Please refactor this to use DiffableDataSource and CompositionalLayout.

File: QuizApp/Quiz/View/QuizzesCategoriesCollectioonViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func set(quizModel: QuizModel) {
+        quizTitle.text = "\(quizModel.name)"
+        let color = quizModel.category.color
+
+        switch quizModel.difficulty {
+        case "EASY":
```
Comment (marinzg) [inline QuizApp/Quiz/View/QuizzesCategoriesCollectioonViewCell.swift:39]: Why isn't this an enum?

**Review by marinzg — CHANGES_REQUESTED** [review]

File: QuizApp/Quiz/View/QuizViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+class QuizViewModel {
+
+    @Published var categories: [SelectedCategoryModel] = []
+    @Published var quizzes: [QuizModel] = []
+
+    private var sport: [QuizModel] = []
+    private var politics: [QuizModel] = []
+    private var movies: [QuizModel] = []
+    private var history: [QuizModel] = []
```
Comment (marinzg) [inline QuizApp/Quiz/View/QuizViewModel.swift:11]: Do you plan to remove these when you connect to the API?

File: QuizApp/Quiz/View/QuizViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    private var router: AppRouterProtocol
+
+    init(router: AppRouterProtocol) {
+        self.router = router
+        setMockData()
+    }
+
+    private func setMockData() {
+        categories = QuizCategory.allCases.map { category in
+            return SelectedCategoryModel(categoryName: category, isSelected: category.id == 0)
```
Comment (marinzg) [inline QuizApp/Quiz/View/QuizViewModel.swift:22]: No need for `return`

**Review by marinzg — CHANGES_REQUESTED** [review]

File: QuizApp/Quiz/View/QuizViewModel.swift (hunk 3 above)
Comment (LorenaL9) [reply QuizApp/Quiz/View/QuizViewModel.swift:11]: Yes, when I connect to the API I will fetch only quizzes in selected category.

File: QuizApp/Quiz/View/QuizzesCategoriesCollectioonViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+class QuizzesCategoriesCollectionViewCell: UICollectionViewCell {
+
+    static let reuseIdentifier = String(describing: QuizzesCategoriesCollectionViewCell.self)
+
+    var quizTitle: UILabel!
+    var quizDescription: UILabel!
+    var quizImage: UIImageView!
+    var score1: UIImageView!
+    var score2: UIImageView!
+    var score3: UIImageView!
```
Comment (dorastipkovic) [inline QuizApp/Quiz/View/QuizzesCategoriesCollectioonViewCell.swift:13]: Use StackView for difficulty images.

**Review by dorastipkovic — CHANGES_REQUESTED** [review]

File: QuizApp/Quiz/View/QuizViewController.swift [hunk 6]
```diff
@@ -39,41 +39,41 @@ class QuizViewController: UIViewController {
...
+            forCellWithReuseIdentifier: CategoryCollectionViewCell.reuseIdentifier)
+        categoryCollectionView.delegate = self
     }
 
     private func bindViewModel() {
         viewModel
             .$categories
             .sink { [weak self] categories in
-                self?.applySnapshotFilter(categories: categories)
+                self?.applyCetegorySnapshot(categories: categories)
```
Comment (dorastipkovic) [inline QuizApp/Quiz/View/QuizViewController.swift:63]: typo

File: QuizApp/Quiz/View/QuizViewController.swift [hunk 7]
```diff
@@ -114,68 +114,80 @@ extension QuizViewController: ConstructViewsProtocol {
...
             heightDimension: .fractionalHeight(1))
+
         let item = NSCollectionLayoutItem(layoutSize: itemSize)
+
         let groupSize = NSCollectionLayoutSize(
             widthDimension: .fractionalWidth(1),
             heightDimension: .absolute(quizGroupHeightDimension))
+
         let group = NSCollectionLayoutGroup.vertical(layoutSize: groupSize, subitems: [item])
+
```
Comment (dorastipkovic) [inline QuizApp/Quiz/View/QuizViewController.swift:182]: You don't need space here. You only need to separate different components.

File: QuizApp.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -90,9 +90,9 @@
...
-		C3238B212892C643005A8DEC /* SelectedCategoryModel.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = SelectedCategoryModel.swift; sourceTree = "<group>"; };
-		C3238B232892CE33005A8DEC /* QuizzesCategoriesCollectioonViewCell.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = QuizzesCategoriesCollectioonViewCell.swift; sourceTree = "<group>"; };
+		C3238B1F2892AE11005A8DEC /* CategoryCollectionViewCell.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CategoryCollectionViewCell.swift; sourceTree = "<group>"; };
+		C3238B212892C643005A8DEC /* SelectableCategoryModel.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = SelectableCategoryModel.swift; sourceTree = "<group>"; };
+		C3238B232892CE33005A8DEC /* QuizCollectioonViewCell.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = QuizCollectioonViewCell.swift; sourceTree = "<group>"; };
```
Comment (dorastipkovic) [inline QuizApp.xcodeproj/project.pbxproj:95]: Typo in file name

**Review by dorastipkovic — CHANGES_REQUESTED** [review]

**Review by dorastipkovic — APPROVED** [review]

**Pull request merged**